R/fileDependency.R

```diff
@@ -0,0 +1,26 @@
+#' Create a file dependency that can be accessed by the client using javascript.
+#' @export
+fileDependency <- function(filename, version = '0.0.1'){
+  htmltools::htmlDependency(
+    name = basename(tools:::file_path_sans_ext(filename)),
+    version = version,
+    src = dirname(filename),
+    attachment = basename(filename)
+  )
```

Comment:
I'm not sure `htmlDependency` is the right substrate to build this on. The reason for this is that htmltools and Shiny assume that once a dependency is satisfied, it's satisfied for good. So in your example, if in a Shiny app, pcs.json changes between one run and the next (and in fact if that's the reason it's changing) then the changes would be ignored.

You could fix this by using a random value for the htmlDependency's name, I guess? That would have an unfortunate side effect of never letting go of old values (or at least references to them)... though I think that's still better than not picking up new values.